components/script/dom/htmlformelement.rs

```diff
@@ -285,9 +287,21 @@ impl HTMLFormElement {
             FormEncType::UrlEncoded => {
                 let mime: mime::Mime = "application/x-www-form-urlencoded".parse().unwrap();
                 load_data.headers.set(ContentType(mime));
-                serialize(form_data.iter().map(|d| (&*d.name, &*d.value)))
+
+                serialize(form_data.iter().map(|d| (&*d.name, match d.value {
+                    FileOrString::StringData(ref s) => String::from(s.clone()),
+                    FileOrString::FileData(ref f) => String::from(f.name().clone())
+                })))
+            }
+            FormEncType::FormDataEncoded => {
+                let boundary = generate_boundary();
+                let mime = "multipart/formdata; boundary=".to_owned() + &boundary;
```

Comment:
Aside from the typo (it's "multipart/form-data"), I hear from @seanmonstar that this is bad practice to use strings instead of actual Mime types that hyper provides.